tests/gem5/se_mode/rvv_intrinsic_tests/test.py

```diff
@@ -0,0 +1,45 @@
+import os
+import re
+import sys
+
+from testlib import *
+
+binaries = [
+    "rvv_branch",
+    "rvv_index",
+    "rvv_matmul",
+    "rvv_memcpy",
+    "rvv_reduce",
+    "rvv_saxpy",
+    "rvv_sgemm",
+    "rvv_strcmp",
+    "rvv_strcpy",
+    "rvv_strlen",
+    "rvv_strlen-fault",
+    "rvv_strncpy",
+]
+
+vlens = [2**x for x in range(7, 15)]
+
+if not (src_dir := os.environ.get("SRCDIR")):
+    print("SRCDIR env var not defined. Exiting.", file=sys.stderr)
+    exit()
+
+for binary in binaries:
+    bin_path = f"{src_dir}/apps/rvv-intrinsic-tests/bin/{binary}"
```

Comment:
These should be on https://resources.gem5.org/ and we should use `obtain_resource`. Please work with @Harshil2107 to get him the source and explain to him how to build the binaries so he can upload them.

Reply:
@Harshil2107 I have opened a repo with the source and a simple Makefile you can use to compile them [here](https://github.com/saul44203/rvv-intrinsic-tests/tree/main). If you have any problem please tell me.

Reply:
We'll get on this :)

Reply:
This is an internal compiler header file used for the rvv intrinsics, so I think your gcc version might be the culprit. Try `riscv64-linux-gnu-gcc-13` if you can.

Reply:
I was able to build the binaries by using `riscv64-linux-gnu-gcc-13` in a docker container. I will upload them to resources. Is it ok if I copy the foler from your repo to gem5-resources as a source for these resources?

Reply:
Great. Yes, you can copy it.

Reply:
I uploaded the binaries and made JSON objects so that we can now use `obtain_resource` to get the binary.

I need a few things:

- What name should I use as author?
- Is there a specific license that I should mention for the resource?
- What would be a good one line description for the resource?

Here are the links to the binaries:
[rvv-strncpy](https://resources.gem5.org/resources/rvv-strncpy?version=1.0.0)  
[rvv-strlen](https://resources.gem5.org/resources/rvv-strlen?version=1.0.0)  
[rvv-memcpy](https://resources.gem5.org/resources/rvv-memcpy?version=1.0.0)  
[rvv-sgemm](https://resources.gem5.org/resources/rvv-sgemm?version=1.0.0)  
[rvv-strcmp](https://resources.gem5.org/resources/rvv-strcmp?version=1.0.0)  
[rvv-index](https://resources.gem5.org/resources/rvv-index?version=1.0.0)  
[rvv-branch](https://resources.gem5.org/resources/rvv-branch?version=1.0.0)  
[rvv-matmul](https://resources.gem5.org/resources/rvv-matmul?version=1.0.0)  
[rvv-reduce](https://resources.gem5.org/resources/rvv-reduce?version=1.0.0)  
[rvv-saxpy](https://resources.gem5.org/resources/rvv-saxpy?version=1.0.0)  
[rvv-strcpy](https://resources.gem5.org/resources/rvv-strcpy?version=1.0.0)  
[rvv-strlen-fault](https://resources.gem5.org/resources/rvv-strlen-fault?version=1.0.0)

Please let me know if I need to update or remake the binaries.

Reply:
Thanks @Harshil2107. I updated the config with `obtain_resource` and all seems to work well.

As for the questions:
- This tests are just minor adaptations of the [riscv-non-isa/rvv-intrinsic-doc examples](https://github.com/riscv-non-isa/rvv-intrinsic-doc/tree/main/examples), so maybe we should reference that.
- I'm not sure about the license. Their repo is under BSD-3 and CC-BY-4.0, so maybe both?
- We could use something like "RISCV-V vector intrinsic {name} example program", where name is "strncpy", "strlen", etc.